ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -468,7 +472,67 @@ public static boolean deserializeDeleteRecord(CrcInputStream crcStream) throws I
             "corrupt data while parsing delete record Expected CRC " + expectedCRC + " Actual CRC " + actualCRC);
         throw new MessageFormatException("delete record data is corrupt", MessageFormatErrorCodes.Data_Corrupt);
       }
-      return isDeleted;
+      return new DeleteRecord(isDeleted);
+    }
+  }
+
+  /**
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   * |         |               |               |               |             |
+   * | version |   AccountId   |  ContainerId  |  DeletionTime |     Crc     |
+   * |(2 bytes)|    (2 byte2)  |   (2 bytes)   |   (4 bytes)   |  (8 bytes)  |
+   * |         |               |               |               |             |
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *  version         - The version of the delete record
+   *
+   *  AccountId     - AccountId that the blob belongs to
+   *
+   *  ContainerId   - ContainerId that the blob belongs to
+   *
+   *  Deletion Time - Time of deletion in secs
+   *
+   *  Crc             - The crc of the delete record
+   *
+   */
+  public static class Delete_Format_V2 {
+
+    public static final int AccountId_Field_Size_In_Bytes = 2;
+    public static final int ContainerId_Field_Size_In_Bytes = 2;
+    public static final int Deletion_Time_Field_Size_In_Bytes = 4;
+
+    private static Logger logger = LoggerFactory.getLogger(Delete_Format_V2.class);
+
+    public static int getDeleteRecordSize() {
+      return Version_Field_Size_In_Bytes + AccountId_Field_Size_In_Bytes + ContainerId_Field_Size_In_Bytes
+          + Deletion_Time_Field_Size_In_Bytes + Crc_Size;
+    }
+
+    public static void serializeDeleteRecord(ByteBuffer outputBuffer, short accountId, short containerId,
+        int deletionTimeInSecs) {
+      int startOffset = outputBuffer.position();
+      outputBuffer.putShort(Delete_Version_V2);
+      outputBuffer.putShort(accountId);
+      outputBuffer.putShort(containerId);
+      outputBuffer.putInt(deletionTimeInSecs);
+      Crc32 crc = new Crc32();
+      crc.update(outputBuffer.array(), startOffset, getDeleteRecordSize() - Crc_Size);
+      outputBuffer.putLong(crc.getValue());
+    }
+
+    public static DeleteRecord deserializeDeleteRecord(CrcInputStream crcStream)
+        throws IOException, MessageFormatException {
+      DataInputStream dataStream = new DataInputStream(crcStream);
+      short accountId = dataStream.readShort();
+      short containerId = dataStream.readShort();
+      int deletionTimeInSecs = dataStream.readInt();
+      long actualCRC = crcStream.getValue();
+      long expectedCRC = dataStream.readLong();
+      if (actualCRC != expectedCRC) {
+        logger.error(
```

Comment:
since there is an exception being thrown, there is no need to log an error. Otherwise, there will be double logging.